### PR TITLE
Improve docker entrypoints

### PR DESCRIPTION
**Pull request by sayanarijit**
- Separate develop from test dependencies
- Use entrypoints to run CI/CD tests
- Fix https://github.com/niteoweb/pyramid-realworld-example-app/issues/88

### CODE REVIEW

File: docker-entrypoints/test.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env nix-shell
+#!nix-shell --argstr type test -i bash /app/shell.nix
+
+# This entrypoint is generally used by the CI/CD pipeline.
+# This runs the given commands in an environment that has all the dependencies
+# from `run` and some additional dependencies to run the tests.
+# e.g. geckodriver for browser tests.
+
+poetry run "$@"
```
Comment (zupo) [inline docker-entrypoints/test.sh]: Why do we need three files with exactly the same content?

Comment (sayanarijit) [reply docker-entrypoints/test.sh]: #!nix-shell --argstr type **test** -i bash /app/shell.nix

Comment (sayanarijit) [reply docker-entrypoints/test.sh]: Here we specify the `type test`

**Comment by sayanarijit** [comment]
I'd prefer using every possible optimization in ci/cd pipeline. Waiting is tedious 😅.

**Comment by sayanarijit** [comment]
Merging this pr.

**Pull request merged**